server/src/main/java/org/apache/druid/server/QueryLifecycle.java

```diff
@@ -347,7 +347,7 @@ public void emitLogsAndMetrics(
 
       if (e != null) {
         statsMap.put("exception", e.toString());
-        if (baseQuery.context().isDebug()) {
+        if (baseQuery.context().isDebug() || e.getMessage() == null) {
```

Comment:
Wont this fill up the logs if we have too many duck exceptions ?

Reply:
Since I do not have a work around yet I think its okay .

Reply:
interesting; reading this I had an idea: how about:
* keeping a `Set<Integer>` around
* compute `h = hash(Throwables.getStackTraceAsString(e))` 
* if its the first occurance of the exception (not in the set) - log it with backtrace along; and also leave `hash = ${h}` behind in some form
* later when the same exception repeats (same hash) -> just log a reference to it's hash so it can be looked up in the logs

Reply:
Overall share of exceptions with null message is very low, about 0.1% among all other meaningful exceptions. 
`org.apache.druid.error.DruidException: null` is not at all useful log. I greped regex `new *.+Exception\(\)` in code, found multiple places, eg `throw new QueryTimeoutException();`, it can be one of them.

More conservative way of printing the log would be to log the base Throwable exception type that DruidException receives. In fact we should print the original exception classname along with message everywhere to make more sense of errors and warning from DruidExceptions.